### PR TITLE
fix: fix loading Button in Badge style

### LINKED ISSUE #23640: When the loading Button added to the Badge, the style is incorrect
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-wcf32)

### Steps to reproduce
-

### What is expected?
-

### What is actually happening?
-

| Environment | Info |
|---|---|
| antd | 3.26.16 |
| React | latest |
| System | MacOS latest |
| Browser | Chrome latest |

### PR DESCRIPTION
**Pull request by yoyo837**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
fix #23640

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix the problem that the badge number style is blocked when put loading button in to the badge      |
| 🇨🇳 Chinese |   修复loading按钮在Badge中遮挡样式的问题        |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/badge/demo/basic.md](https://github.com/ant-design/ant-design/blob/fix-loading-btn-in-badge/components/badge/demo/basic.md)

### CODE REVIEW

**Comment by afc163** [comment]
看看修改完什么效果。

**Comment by yoyo837** [comment]
image

咋CI不过呢，本地可以。

**Comment by zombieJ** [comment]
本地用 npm 安装一下，覆盖测试用例。 cnpm 把 react-resizable 加了一个 bug version

**Comment by yoyo837** [comment]
Cannot find module '@jest/globals' 这玩意儿哪里配置

File: components/badge/style/index.less [hunk 1]
```diff
@@ -51,6 +51,7 @@
     position: absolute;
     top: 0;
     right: 0;
+    z-index: 1;
```
Comment (afc163) [inline components/badge/style/index.less:54]: 这样上面的 `z-index: @zindex-badge;` 是不是就无效了？

**Comment by yoyo837** [comment]
会影响fixed columns #20751

File: components/style/themes/default.less [hunk 2]
```diff
@@ -265,7 +265,7 @@
 @layout-trigger-color-light: @text-color;
 
 // z-index list, order by `z-index`
-@zindex-badge: auto;
+@zindex-badge: 1;
```
Comment (afc163) [inline components/style/themes/default.less:268]: 改过很多次，需要 blame 看看之前解决的问题会不会再出现。

**Comment by yoyo837** [comment]
affix  #15782

**Comment by yoyo837** [comment]
"antd": "git://github.com/ant-design/ant-design.git#fix-loading-btn-in-badge"

codesandbox 不能这么写么？

**Comment by yoyo837** [comment]
preview : https://codesandbox.io/s/admiring-margulis-5oruv?file=/index.js

**Comment by zombieJ** [comment]
我修了一下 3.x branch 的 CI，可以 rebase 一下。

**Comment by yoyo837** [comment]
> 我修了一下 3.x branch 的 CI，可以 rebase 一下。

依旧是本地可以，不过这次CI是moment相关的错误

**Comment by yoyo837** [comment]
Done

File: components/style/themes/default.less (hunk 2 above)
Comment (zombieJ) [reply components/style/themes/default.less:268]: @yoyo837, 这个有看过上下文吗？

Comment (yoyo837) [reply components/style/themes/default.less:268]: 应该没破坏之前的效果。

**Pull request merged**